runtime/dom-particle.js

```diff
@@ -126,25 +119,13 @@ class DomParticle extends XenStateMixin(Particle) {
       slot.render({});
     }
   }
-  _initializeRender(slot) {
-    let template = this.getTemplate(slot.slotName);
-    this._findHandlerNames(template).forEach(name => {
-      slot.clearEventHandlers(name);
-      slot.registerEventHandler(name, eventlet => {
-        if (this[name]) {
-          this[name](eventlet, this._state, this._views);
-        }
-      });
-    });
-    return template;
-  }
-  _findHandlerNames(html) {
-    let handlers = new Map();
-    let re = /on-.*?=\"([^\"]*)"/gmi;
-    for (let m=re.exec(html); m; m=re.exec(html)) {
-      handlers.set(m[1], true);
+  fireEvent(slotName, {handler, data}) {
+    if (this[handler]) {
+      this[handler]({data});
```

Comment:
this[handler]({data}, this._state);

would fix the Chooser, this is how the handler would bound originally:
https://github.com/PolymerLabs/arcs/blob/master/runtime/dom-particle.js#L135